tests/publiccloud/azure_cli.pm

```diff
@@ -17,16 +17,36 @@ use mmapi 'get_current_job_id';
 use utils qw(zypper_call script_retry);
 use version_utils 'is_sle';
 use registration qw(add_suseconnect_product get_addon_fullname);
+use strict;
+use warnings;
+use publiccloud::utils;
+use publiccloud::ssh_interactive 'select_host_console';
+
+our $root_dir = '/root';
 
 sub run {
     my ($self, $args) = @_;
     select_serial_terminal;
     my $job_id = get_current_job_id();
+    set_var 'PUBLIC_CLOUD_PROVIDER' => 'AZURE';
+    my $ins_pro = $self->provider_factory();
+
+    my $azure_repo = get_required_var('PY_AZURE_REPO');
+    my $backports_repo = get_required_var('PY_BACKPORTS_REPO');
+    my $cloud_tools_repo = get_required_var('CLOUD_TOOLS_REPO');
+
+
+    # install cloud_tools packages in sles instance and validate azure cli test 
+    if (($azure_repo ne '') && ($backports_repo ne '') && ($cloud_tools_repo ne '')) {
+        create_sle_vm($azure_repo,$backports_repo,$cloud_tools_repo,$ins_pro);
+    }
 
     # If 'az' is preinstalled, we test that version
     if (script_run("which az") != 0) {
         add_suseconnect_product(get_addon_fullname('pcm'), (is_sle('=12-sp5') ? '12' : undef));
+        add_suseconnect_product(get_addon_fullname('pcm'), (is_sle('=15-sp4') ? '15' : undef));
```

Comment:
```
my $version;
$version = '12' if is_sle('=12-sp5');
$version = '15' is_sle('=15-sp4');
add_suseconnect_product(get_addon_fullname('pcm'), $version);
```
   otherwise you calling  `add_suseconnect_product` two times which is bad